snaptrade/api_client/snaptrade_api_client.py

```diff
@@ -132,10 +132,17 @@ def _make_request(self, endpoint_name, data=None, path_params=None, query_params
         if self.debug_response:
             return response
 
-        if response.content:
+        if response.status_code == 502:
+            error_message = dict(status_code=502, detail="Failed to connect to api server", code="0000")
+            if self.return_response_as_dict:
+                return error_message
+            else:
+                return SnapTradeUtils.convert_to_simple_namespace(error_message)
+
+        try:
             data = response.json()
-        else:
-            data = dict(status_code=response.status_code, detail="No content returned")
+        except:
+            data = dict(status_code=response.status_code, detail=f"{str(response.content)}")
```

Comment:
Should this have a `code` field too?

Reply:
I pushed some code to handle 204 no content more explicitly and added a coder to the other error.